syntax = "proto3";

package event;

service EventService {
  rpc CreateEvent(CreateEventRequest) returns (CreateEventResponse);
  rpc UpdateEvent(UpdateEventRequest) returns (UpdateEventResponse);
  rpc ListEvents(ListEventsRequest) returns (ListEventsResponse);
  rpc Status(StatusRequest) returns (StatusResponse);
}

message CreateEventRequest {
  string title = 1;
  string description = 2;
  string date = 3;
}

message CreateEventResponse {
  string id = 1;
}

message UpdateEventRequest {
  string id = 1;
  string title = 2;
  string description = 3;
  string date = 4;
}

message UpdateEventResponse {
  bool success = 1;
}

message ListEventsRequest {}

message Event {
  string id = 1;
  string title = 2;
  string description = 3;
  string date = 4;
}

message ListEventsResponse {
  repeated Event events = 1;
}

message StatusRequest {}

message StatusResponse {
  string status = 1;
}
